File: 01-Languages/Dart/05-Control-Flow/Exercises/exercise.dart
// Exercise: FizzBuzz using a Dart 3 switch EXPRESSION (not if/else if).
// Write a function fizzbuzz(int n) that returns:
//   "FizzBuzz" if divisible by both 3 and 5
//   "Fizz"     if divisible by 3 only
//   "Buzz"     if divisible by 5 only
//   otherwise, the number itself as a String
// Use `switch (n) { ... }` as an EXPRESSION with `when` guard clauses (Dart 3 pattern syntax).

String fizzbuzz(int n) {
  // TODO: implement using a switch expression, e.g.:
  // return switch (n) {
  //   _ when n % 15 == 0 => 'FizzBuzz',
  //   ...
  // };
  return '';
}

void main() {
  for (var i = 1; i <= 15; i++) {
    print(fizzbuzz(i));
  }
}

File: 01-Languages/PHP/05-Control-Flow/Exercises/exercise.php
<?php
// Exercise: FizzBuzz using `match`, PHP 8's expression-based alternative to switch.
// Write a function fizzbuzz(int $n): string that returns:
//   "FizzBuzz" if $n is divisible by both 3 and 5
//   "Fizz"     if divisible by 3 only
//   "Buzz"     if divisible by 5 only
//   otherwise, the number itself, as a string
// Use `match (true) { ... }` (the same pattern shown in example.php) rather than if/elseif.

function fizzbuzz(int $n): string {
    // TODO: implement using match (true) { condition => result, ... }
    return "";
}

for ($i = 1; $i <= 15; $i++) {
    echo fizzbuzz($i) . "\n";
}
